File: app/src/main/java/com/clumob/segment/MainActivity.java
package com.clumob.segment;

import android.content.Context;
import android.os.Bundle;
import androidx.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.clumob.segment.controller.SegmentController;
import com.clumob.segment.controller.SegmentControllerImpl;
import com.clumob.segment.controller.SegmentInfo;
import com.clumob.segment.controller.Storable;
import com.clumob.segment.manager.Segment;
import com.clumob.segment.manager.SegmentView;
import com.clumob.segment.manager.SegmentViewHolderFactory;
import com.clumob.segment.manager.SegmentManager;
import com.clumob.segment.manager.SegmentNavigation;
import com.clumob.segment.manager.SegmentViewHolder;
import com.clumob.segment.support.appcompact.SegmentAppCompatActivity;

public class MainActivity extends SegmentAppCompatActivity  {


    @Override
    public Segment provideSegment(SegmentInfo segmentInfo) {
        return new Segment(segmentInfo, new SegmentControllerImpl(segmentInfo.getArguments(), null), new SegmentViewHolderFactory() {
            @Override
            public SegmentViewHolder<?, ?> create(Context context, LayoutInflater layoutInflater, @Nullable ViewGroup parentView) {
                return new TestSegmentScreenHolder(context, layoutInflater, parentView);
            }
        });
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SegmentView segmentView = new SegmentView(this);
        SegmentInfo segmentInfo = new SegmentInfo(1, null);
        segmentView.setSegment(new Segment(segmentInfo, new SegmentControllerImpl(segmentInfo.getArguments(), null), new SegmentViewHolderFactory() {
            @Override
            public SegmentViewHolder<?, ?> create(Context context, LayoutInflater layoutInflater, @Nullable ViewGroup parentView) {
                return new TestSegmentScreenHolder(context, layoutInflater, parentView);
            }
        }));
        setContentView(segmentView);
//        getSegmentManager().getNavigation().navigateToScreen(new SegmentInfo<Storable, Storable>(1, null));
    }

    @Override
    public void setSegmentView(View view) {
        setContentView(view);
    }

    @Override
    public SegmentNavigation createSegmentNavigation(final SegmentManager segmentManager) {
        return new SegmentNavigation(segmentManager) {

        };
    }

}

File: app/src/main/java/com/clumob/segment/MainActivity.kt
package com.clumob.segment

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.clumob.segment.activity.SegmentAppCompatActivity
import com.clumob.segment.controller.SegmentInfo
import com.clumob.segment.manager.*
import com.clumob.segment.view.SegmentViewProvider

class MainActivity : SegmentAppCompatActivity() {
    override fun provideSegment(segmentInfo: SegmentInfo): Segment {
        val context = this
        val segment = Segment(SubSegmentController(), object : SegmentViewProvider {
            override fun create(parent: ViewGroup?, viewType: Int): SegmentViewHolder {
                return TestSegmentViewHolder(context, LayoutInflater.from(context), parent)
            }
        })
        segment.bindSegmentInfo(segmentInfo)
        return segment
    }

    @SuppressLint("NewApi")
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        segmentManager!!.navigation.navigateToScreen(SegmentInfo(1,null))
    }

    override fun setSegmentView(view: View) {
        setContentView(view)
    }

    override fun onRestart() {
        super.onRestart()
        segmentManager!!.navigation.navigateToScreen(SegmentInfo(1,null))
    }

    override fun onPause() {
        Log.d("SEGMENT"," Activity Pause Begin")
        super.onPause()
        Log.d("SEGMENT"," Activity Pause end")
    }

    override fun onStop() {
        Log.d("SEGMENT"," Activity Stop")
        super.onStop()
        Log.d("SEGMENT"," Activity Stop End")
    }

    override fun onDestroy() {
        Log.d("SEGMENT"," Activity Destroy Start")
        super.onDestroy()
        Log.d("SEGMENT"," Activity Destroy End")
    }

    override fun createSegmentNavigation(segmentManager: SegmentManager): SegmentNavigation {
        return SegmentNavigation(segmentManager)
    }
}